packages/app-builder-lib/src/node-module-collector/index.ts

```diff
@@ -3,14 +3,34 @@ import { PnpmNodeModulesCollector } from "./pnpmNodeModulesCollector"
 import { YarnNodeModulesCollector } from "./yarnNodeModulesCollector"
 import { detect, PM, getPackageManagerVersion } from "./packageManager"
 import { NodeModuleInfo } from "./types"
+import { exec } from "builder-util"
+
+async function isPnpmProjectHoisted(rootDir: string) {
+  const command = process.platform === "win32" ? "pnpm.cmd" : "pnpm"
+  const config = await exec(command, ["config", "list"], { cwd: rootDir, shell: true })
+  const lines = config
+    .split("\n")
```

Comment:
If you split like this on Windows, `\r` may remain in the result, so it's necessary to `trim` the final output.